cql3/untyped_result_set.hh

```diff
@@ -41,36 +41,61 @@
  */
 #include <unordered_map>
 #include <optional>
+#include <seastar/core/sharded.hh>
 #include "bytes.hh"
 #include "types.hh"
 #include "types/map.hh"
 #include "types/list.hh"
 #include "types/set.hh"
+#include "serializer.hh"
+#include "bytes_ostream.hh"
 #include "transport/messages/result_message_base.hh"
 #include "column_specification.hh"
-#include "absl-flat_hash_map.hh"
+#include "query-result.hh"
 
 #pragma once
 
+namespace query {
+    class result;
+    class partition_slice;
+    // duplicate template def. But avoids a huge include chain
+    using result_bytes_view = ser::buffer_view<bytes_ostream::fragment_iterator>;
+}
+
 namespace cql3 {
+namespace selection {
+    class selection;
+}
+
+class untyped_result_set;
+class result;
+class metadata;
 
 class untyped_result_set_row {
+public:
+    using view_type = query::result_bytes_view;
+    using opt_view_type = std::optional<view_type>;
+    using view_holder = std::variant<std::monostate, view_type, bytes>;
 private:
-    const std::vector<lw_shared_ptr<column_specification>> _columns;
-    using map_t = flat_hash_map<sstring, bytes_opt>;
-    const map_t _data;
+    friend class untyped_result_set;
+    using index_map = std::unordered_map<std::string_view, size_t>;
+    using data_views = std::vector<view_holder>;
+    using index_map_ptr = ::shared_ptr<index_map>;
```

Comment:
could this be a lw_shared_ptr instead of ::shared_ptr?

Reply:
I guess. The types are known, so... Could even be a ref + unique_ptr in parent object.